libs/tokens/src/utils/fetchTokenFromBlockchain.ts

```diff
@@ -0,0 +1,29 @@
+import { getContract } from '@cowprotocol/common-utils'
+import { Erc20, Erc20Abi } from '@cowprotocol/abis'
+import type { JsonRpcProvider } from '@ethersproject/providers'
+import { getAddress } from '@ethersproject/address'
+import { TokenInfo } from '@uniswap/token-lists'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+export async function fetchTokenFromBlockchain(
+  tokenAddress: string,
+  chainId: SupportedChainId,
+  provider: JsonRpcProvider
+): Promise<TokenInfo> {
+  const formattedAddress = getAddress(tokenAddress)
+  const erc20Contract = getContract(formattedAddress, Erc20Abi, provider) as Erc20
+
+  const [name, symbol, decimals] = await Promise.all([
+    erc20Contract.callStatic.name(),
+    erc20Contract.callStatic.symbol(),
+    erc20Contract.callStatic.decimals(),
+  ])
```

Comment:
2 minor points, as I don't expect the be used often or at all to be honest:

1. Don't we have a service in the project that does this already?
2. What about tokens that encode the strings differently? (Can only think of DAI and MAKER)

Reply:
1. We had, but I deleted it: https://github.com/cowprotocol/cowswap/blob/develop/apps/cowswap-frontend/src/legacy/hooks/useTokenLazy.ts
2. The legacy hook was considered the case, but I don't fully understand when it happens? Checked DAI and MAKER tokens and there is no problem. I would assume that `@ethersproject/contracts` parse different encoding under the hood.